Update GitHub Actions workflows to support Python 3.11

sbshah97 opened the pull request:
Update the GitHub Actions workflows and documentation to support the latest stable Python version (3.11).

* **README.md**
  - Update the "Requirements" section to include Python 3.11.

* **setup.py**
  - Update the `classifiers` list to include Python 3.11.

* **.github/workflows/publish-distribution.yaml**
  - Update the `python-version` to '3.11' in the "Set up Python" step.

* **.github/workflows/publish-documentation.yaml**
  - Update the `python-version` to '3.11' in the "Set up Python" step.

* **.github/workflows/test-build.yaml**
  - Add '3.11' to the `python-version` matrix in the "mock-api-test" job.

sbshah97 committed:
Update GitHub Actions workflows to support Python 3.11

Update the GitHub Actions workflows and documentation to support the latest stable Python version (3.11).

* **README.md**
  - Update the "Requirements" section to include Python 3.11.

* **setup.py**
  - Update the `classifiers` list to include Python 3.11.

* **.github/workflows/publish-distribution.yaml**
  - Update the `python-version` to '3.11' in the "Set up Python" step.

* **.github/workflows/publish-documentation.yaml**
  - Update the `python-version` to '3.11' in the "Set up Python" step.

* **.github/workflows/test-build.yaml**
  - Add '3.11' to the `python-version` matrix in the "mock-api-test" job.

tim-smar commented:
@sbshah97 It appears the linter is failing due to a [new pylint rule](https://pylint.readthedocs.io/en/v3.3.3/user_guide/messages/refactor/too-many-positional-arguments.html). I started a [PR](https://github.com/smartsheet/smartsheet-python-sdk/pull/58/files) to resolve that issue, but it failed on python 3.7 which is past its end of life. Since you've already started to address python version issues here, I closed my PR. If you make the same change I referenced, this PR should be able to move forward. You may need to drop support for 3.7 and 3.8 before all the checks pass.

sbshah97 committed:
Update Python versions and resolve linter issue

Update project to drop support for Python 3.7 and 3.8 and resolve linter issue.

* **setup.py**
  - Remove support for Python 3.7 and 3.8 from the `classifiers` list.
  - Remove the `enum34` dependency from the `REQUIRES` list as this checks for python versions below 3.4.

* **README.md**
  - Update the "Requirements" section to mention compatibility with Python 3.9, 3.10, and 3.11.

* **.github/workflows/test-build.yaml**
  - Remove Python 3.7 and 3.8 from the `matrix` list under the `mock-api-test` job.

sbshah97 committed:
Merge pull request #3 from sbshah97/update-python-versions

Update Python versions and resolve linter issue
